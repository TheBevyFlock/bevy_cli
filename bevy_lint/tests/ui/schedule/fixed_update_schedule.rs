#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::fixed_update_schedule)]

use bevy::prelude::*;

fn main() {
    let mut app = App::new();

    app.add_plugins(DefaultPlugins)
        //~v ERROR: the `FixedUpdate` schedule is disallowed
        .add_systems(FixedUpdate, hello_world);

    // Ensure the lint can be muted by annotating the expression.
    app.add_systems(
        #[expect(bevy::fixed_update_schedule)]
        FixedUpdate,
        hello_world,
    );

    app.run();
}

fn hello_world() {
    println!("hello world!");
}
