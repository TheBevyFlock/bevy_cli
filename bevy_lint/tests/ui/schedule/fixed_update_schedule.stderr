error: the `FixedUpdate` schedule is disallowed
  --> tests/ui/schedule/fixed_update_schedule.rs:12:22
   |
12 |         .add_systems(FixedUpdate, hello_world);
   |                      ^^^^^^^^^^^
   |
note: the lint level is defined here
  --> tests/ui/schedule/fixed_update_schedule.rs:3:9
   |
 3 | #![deny(bevy::fixed_update_schedule)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

